### DISCUSSION TITLE
What cases will you use the map argument with @unique?

### DISCUSSION THREAD

**Discussion by masashi-sutou** [prisma/prisma#11232] [Q&A] [answered]
- It is blank on description of `@unique`'s map argument  in prisma document.
- What cases will you use the map argument with @unique?
- Basically, in most cases you don't need to use the map argument?

What cases will you use the map argument with @unique? |
:-: | 
<img src=https://user-images.githubusercontent.com/6649643/149850008-02c0f608-a7c0-4b43-a695-ee90ca89259e.png width=580/> |

**Comment by austincrim** [comment, marked as answer]
Hey @masashi-sutou 👋 ,

When using the `@unique` directive, the `map` argument allows you to provide the name for the unique constraint that's created in the underlying database instead of allowing Prisma to generate a name for you.

I would say a majority of users will not need to use it, but it is a useful escape hatch if you ever need to directly control the naming.

I'll make sure our docs team knows about the blank description field, thanks for pointing that out!
Let me know if you have any more questions!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
